crowbar_framework/app/models/ntp_service.rb

```diff
@@ -110,8 +110,12 @@ def skip_unchanged_node?(node_name, old_role, new_role)
     # if the node changed roles, then we need to apply
     return false if node_changed_roles?(node_name, old_role, new_role)
 
-    # if attributes have changed, we need to run
-    return false if node_changed_attributes?(node_name, old_role, new_role)
+    # if attributes have changed except external_servers, we need to apply
+    return false if relevant_attributes_changed_if_roles?(node_name, old_role, new_role,
+      ["external_servers"], ["ntp-client"])
+
+    # if the servers has changed, we need to apply
+    return false if old_role.elements["ntp-server"].sort != new_role.elements["ntp-server"].sort
```

Comment:
Last comment on this: I would actually make this the second test (before `node_changed_roles?`) as this one is very simple and generic, and doesn't depend on the node itself.